### PR TITLE
feat(login): Login page

### PR DESCRIPTION
**Pull request by mmarchal**
Issue : https://github.com/XPEHO/xpeapp_admin/issues/1

Feature video : 

https://github.com/XPEHO/xpeapp_admin/assets/25179165/40b876c6-2d40-4ced-a5c5-b1c626cf1d80

### CODE REVIEW

File: lib/env/extensions/user.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    } else {
+      String email = this.email!;
+      String nameBeforeAt = email.substring(0, email.indexOf('@'));
+      if (nameBeforeAt.contains('.')) {
+        List<String> nameBeforeAtSplit =
+            nameBeforeAt.split('.'); // ['john', 'doe']
+        // Upper case first letter of first name and last name
+        String firstName = nameBeforeAtSplit[0].capitalize();
+        String lastName = nameBeforeAtSplit[1].toUpperCase();
+        return '$firstName $lastName';
```
Comment (PiotrFLEURY) [inline lib/env/extensions/user.dart:17]: should be parsed using a regex

File: web/manifest.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+    "name": "xpeapp_admin",
+    "short_name": "xpeapp_admin",
+    "start_url": ".",
+    "display": "standalone",
+    "background_color": "#0175C2",
+    "theme_color": "#0175C2",
+    "description": "A new Flutter project.",
```
Comment (PiotrFLEURY) [inline web/manifest.json:8]: don't forget to update it

**Review by PiotrFLEURY — APPROVED** [review]
Globally OK

Windows platform can be deleted

Please consider comments

File: .github/workflows/flutter.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+on: pull_request
+
+name: Flutter Test
+jobs:
+  build:
+    name: Flutter pub get, format, analyze, and test
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Create firebase config file
```
[the remark this answers is not in the document]
Comment (mmarchal) [reply .github/workflows/flutter.yaml:10]: steps:
    - uses: subosito/flutter-action@v2
      with:
        flutter-version: '3.13.1'
        channel: 'stable'

    - name: Create firebase config file
      uses: actions/checkout@v3
    - run: |
        echo ${{ secrets.FIREBASE_CONFIGURATION }} | base64 -d > lib/firebase_options.dart
        
    - name: Flutter pub get, format, analyze, and test
      run: |
        dart format lib/firebase_options.dart
        flutter pub get
        dart format --set-exit-if-changed .
        flutter analyze
        flutter test
        
That ?

File: web/manifest.json (hunk 2 above)
Comment (mmarchal) [reply web/manifest.json:8]: Corrected !

File: lib/env/extensions/user.dart (hunk 1 above)
Comment (mmarchal) [reply lib/env/extensions/user.dart:17]: You have an example of regex used ?

Comment (mmarchal) [reply lib/env/extensions/user.dart:17]: New version 
```
extension GetNameOfUser on User {
  String getName() {
    // Check if displayName is not null
    if (displayName != null) {
      return displayName!;
    } else {
      String email = this.email ?? '';

      // Split the email by @ and then by . to get the name parts
      List<String> nameParts = email.split('@')[0].split('.');

      if (nameParts.isNotEmpty) {
        // Set the first name to the first part of the nameParts and capitalize it
        String firstName = nameParts[0].capitalize();
        String lastName =
            nameParts.length > 1 ? nameParts[1].toUpperCase() : '';

        return '$firstName $lastName'.trim();
      } else {
        // If the nameParts is empty, capitalize the first letter of the email
        return email[0].toUpperCase() + email.substring(1);
      }
    }
  }
}
```

**Pull request merged**